orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py

```diff
@@ -327,12 +327,27 @@ def _get_exported_model(self, input_schema: ORTModelInputOutputSchemaType, *inpu
 
         # Setup dynamic axes for onnx model
         self._input_info = _io.parse_inputs_for_onnx_export(self._module_parameters, None, input_schema, inputs, kwargs)
+        do_deep_copy = self._runtime_options.do_deepcopy_before_model_export and _io.can_module_be_deep_cloned(
+            self._original_module, self._device
+        )
+        if not do_deep_copy:
+            if self._runtime_options.do_deepcopy_before_model_export:
+                self._logger.warning(
+                    "This model will NOT be deep copied (or pickled) as the user requested, "
+                    "Compute will continue, but unexpected results may occur!"
```

Comment:
```suggestion
                    "Compute will continue. If unexpected results occur like (1 or 2 examples), set env variable ORTMODULE_DO_DEEPCOPY_BEFORE_MODEL_EXPORT=1"
```

nit: "but unexpected results may occur" does not explain what to expect and can scare the user.

Reply:
Invite Bing Chat to refine the error message, updated.